File: 2-Ad-Hoc/ex1943.c
#include <stdio.h>

int main(){
    //variaveis
    int n, i, tops[7] = {1, 3, 5, 10, 25, 50, 100};

    //entrada
    scanf("%d", &n);

    //processamento & saida
    for(i = 0; i < 7; i++){
        if(n <= tops[i]){
            printf("Top %d\n", tops[i]);
            break;
        }
    }
    return 0;
}

File: 2-Ad-Hoc/Ex1943.java
import java.util.Scanner;

public class Ex1943 { // para ser aceito no URI mude o nome da classe para Main
    public static void main(String args[]) {
        /* variaveis */
        Scanner scanner = new Scanner(System.in);
        int n, tops[] = { 1, 3, 5, 10, 25, 50, 100 };

        /* entrada */
        n = scanner.nextInt();

        /* processamento & saida */
        for (int i = 0; i < tops.length; i++) {
            if (n <= tops[i]) {
                System.out.println("Top " + tops[i]);
                break;
            }
        }

        scanner.close();
    }
}